eureka-cli/cmd/deployUi.go

```diff
@@ -46,49 +44,38 @@ var deployUiCmd = &cobra.Command{
-	var stripesBranch = internal.GetStripesBranch(deployUiCommand, defaultStripesBranch)
+	slog.Info(deployUiCommand, internal.GetFuncName(), "### CLONING & UPDATING UI ###")
 
+	slog.Info(deployUiCommand, internal.GetFuncName(), fmt.Sprintf("Cloning %s from a %s branch", platformCompleteDir, defaultStripesBranch))
 	outputDir := fmt.Sprintf("%s/%s", internal.DockerComposeWorkDir, platformCompleteDir)
+	stripesBranch := internal.GetStripesBranch(deployUiCommand, defaultStripesBranch)
 	internal.GitCloneRepository(deployUiCommand, enableDebug, internal.PlatformCompleteRepositoryUrl, stripesBranch, outputDir, false)
 
+	slog.Info(deployUiCommand, internal.GetFuncName(), fmt.Sprintf("Pulling updates for %s from origin", platformCompleteDir))
+	internal.GitResetHardPullFromOriginRepository(deployUiCommand, enableDebug, internal.PlatformCompleteRepositoryUrl, defaultStripesBranch, outputDir)
+
+	slog.Info(deployUiCommand, internal.GetFuncName(), "### ACQUIRING KEYCLOAK MASTER ACCESS TOKEN ###")
+	masterAccessToken := internal.GetKeycloakMasterRealmAccessToken(createUsersCommand, enableDebug)
+
 	for _, value := range internal.GetTenants(deployUiCommand, enableDebug, false) {
 		mapEntry := value.(map[string]interface{})
 		tenant := mapEntry["name"].(string)
 
-		if !slices.Contains(viper.GetStringSlice(internal.TenantsKey), tenant) {
+		if !internal.HasTenant(tenant) || !internal.DeployUi(tenant) {
 			continue
 		}
 
-		slog.Info(deployUiCommand, "### UPDATING KEYCLOAK PUBLIC CLIENT", "")
-		masterAccessToken := internal.GetKeycloakMasterRealmAccessToken(createUsersCommand, enableDebug)
+		slog.Info(deployUiCommand, internal.GetFuncName(), "### UPDATING KEYCLOAK PUBLIC CLIENT")
 		internal.UpdateKeycloakPublicClientParams(deployUiCommand, enableDebug, tenant, masterAccessToken)
 
-		slog.Info(deployUiCommand, "### COPYING PLATFORM COMPLETE UI CONFIGS ###", "")
-		files, err := filepath.Glob("eureka-tpl/*")
-		if err != nil {
-			// Handle error if the pattern doesn't match any files
-			internal.LogErrorPanic(deployUiCommand, fmt.Sprintf("Failed to glob files: %v", err.Error()))
-		}
+		slog.Info(deployUiCommand, internal.GetFuncName(), "### COPYING PLATFORM COMPLETE UI CONFIGS ###")
+		internal.RunCommandFromDir(deployUiCommand, exec.Command("cp", "-R", "-f", "eureka-tpl/*", "."), outputDir)
 
-		if len(files) > 0 {
-			// Append the destination directory to the list of files
-			args := append([]string{"-R", "-f"}, files...)
-			args = append(args, ".")
-			internal.RunCommandFromDir(deployUiCommand, exec.Command("cp", args...), outputDir)
-
-			if stripesBranch == defaultStripesBranch {
-				slog.Info(deployUiCommand, "### PREPARING PLATFORM COMPLETE UI CONFIGS ###", "")
-				internal.PrepareStripesConfigJs(deployUiCommand, outputDir, tenant)
-				internal.PreparePackageJson(deployUiCommand, outputDir, tenant)
-			}
-		} else {
-			slog.Info("No files matched in eureka-tpl/*. Not copying.")
-		}
```

Comment:
This does the equivalent of `/*` and also doesn't fail if the directory `eureka-tpl` doesn't exist or is empty. Please see comment above.

Reply:
> This does the equivalent of `/*` and also doesn't fail if the directory `eureka-tpl` doesn't exist or is empty. Please see comment above.

I found it silently avoiding copying all together and building the Okapi-based UI, I'll retest it again tomorrow

Reply:
If this code works on Windows, I think we have something cross-platform. See what you think.